src/backend/distributed/utils/maintenanced.c

```diff
@@ -300,6 +302,8 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 
 		CHECK_FOR_INTERRUPTS();
 
+		Assert(myDbData->workerPid == MyProcPid);
```

Comment:
Isolation tests still hitting this assertion

Reply:
That should definitely be investigated then.